src/rust-crypto/rust-crypto.ts

```diff
@@ -610,15 +610,17 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
      * Implementation of {@link CryptoApi#isCrossSigningReady}
      */
     public async isCrossSigningReady(): Promise<boolean> {
-        const { publicKeysOnDevice, privateKeysInSecretStorage, privateKeysCachedLocally } =
-            await this.getCrossSigningStatus();
+        const { privateKeysInSecretStorage, privateKeysCachedLocally } = await this.getCrossSigningStatus();
         const hasKeysInCache =
             Boolean(privateKeysCachedLocally.masterKey) &&
             Boolean(privateKeysCachedLocally.selfSigningKey) &&
             Boolean(privateKeysCachedLocally.userSigningKey);
 
-        // The cross signing is ready if the public and private keys are available
-        return publicKeysOnDevice && (hasKeysInCache || privateKeysInSecretStorage);
+        const identity = await this.getOwnIdentity();
+
+        // The cross signing is ready if cross-signing is set up, the identity is trusted,
+        // and private keys are available somewhere.
```

Comment:
> The cross signing is ready if cross-signing is set up

This seems a bit like a tautology? What exactly does "cross-signing is set up" mean? I'm not sure it's adding any value here.

```suggestion
        // Cross signing is ready if the public identity is trusted, and the private keys 
        // are either cached, or accessible via secret-storage.
```

Reply:
Just that the public identity could be null, if you never set it up. But yes maybe not worth it